experimental/dds/tree2/src/simple-tree/toMapTree.ts

```diff
@@ -0,0 +1,223 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IFluidHandle } from "@fluidframework/core-interfaces";
+import { assert } from "@fluidframework/core-utils";
+
+import {
+	type FieldKey,
+	type MapTree,
+	type TreeFieldStoredSchema,
+	type TreeNodeSchemaIdentifier,
+	type TreeNodeStoredSchema,
+	type TreeTypeSet,
+} from "../core";
+import { brand, fail } from "../util";
+import {
+	allowsValue,
+	type ContextuallyTypedNodeData,
+	getFieldKind,
+	getFieldSchema,
+	getPossibleTypes,
+	getPrimaryField,
+	isFluidHandle,
+	Multiplicity,
+	type PrimitiveValue,
+	type TreeDataContext,
+	type TreeFieldSchema,
+	type TreeNodeSchema,
+} from "../feature-libraries";
+import { type TreeField, type TypedNode } from "./types";
+
```

Comment:
TODO: add a comment explaining the approach taken and how we need to change it to support new kinds of input -> output type mappings